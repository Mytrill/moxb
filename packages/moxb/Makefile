all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
