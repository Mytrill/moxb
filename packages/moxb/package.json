{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
