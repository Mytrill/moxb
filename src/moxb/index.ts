export * from './impl';
export * from './meteor';
export * from './types';
