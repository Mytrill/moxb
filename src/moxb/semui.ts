export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
