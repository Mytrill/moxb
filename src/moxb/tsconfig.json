{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
