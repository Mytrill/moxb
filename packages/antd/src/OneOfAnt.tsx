import { observer } from 'mobx-react';
import * as React from 'react';
import { Select, Radio, Form } from 'antd';
import { SelectProps } from 'antd/lib/select';
import { RadioProps } from 'antd/lib/radio';
import { BindAntProps, labelWithHelp, parseProps } from './BindAnt';
import { OneOf } from '@moxb/moxb';
import { FormItemProps } from 'antd/lib/form/FormItem';

@observer
export class OneOfAnt extends React.Component<BindAntProps<OneOf> & RadioProps> {
    render() {
        const { operation, invisible, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Radio.Group onChange={(selectionValue: any) => operation.setValue(selectionValue)} {...props}>
                {operation.choices.map(opt => (
                    <Radio key={opt.value} value={opt.value} checked={opt.value === operation.value}>
                        {opt.label}
                    </Radio>
                ))}
            </Radio.Group>
        );
    }
}

@observer
export class OneOfFormAnt extends React.Component<BindAntProps<OneOf> & RadioProps & FormItemProps> {
    render() {
        const { operation, label, invisible } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Form.Item label={labelWithHelp(label != null ? label : operation.label, operation.help)}>
                <OneOfAnt operation={operation} />
            </Form.Item>
        );
    }
}

@observer
export class OneOfSelectAnt extends React.Component<BindAntProps<OneOf> & SelectProps> {
    render() {
        const { operation, invisible, value, placeholder, ...props } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Select
                onChange={(selectionValue: any) => operation.setValue(selectionValue)}
                defaultValue={operation.value}
                value={value}
                placeholder={placeholder}
                mode="default"
                {...props}
            >
                {operation.choices.map(opt => (
                    <Select.Option key={opt.value} value={opt.value}>
                        {opt.label}
                    </Select.Option>
                ))}
            </Select>
        );
    }
}

@observer
export class OneOfSelectFormAnt extends React.Component<BindAntProps<OneOf> & RadioProps & FormItemProps> {
    render() {
        const { operation, label, invisible } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        return (
            <Form.Item label={labelWithHelp(label != null ? label : operation.label, operation.help)}>
                <OneOfSelectAnt operation={operation} />
            </Form.Item>
        );
    }
}
