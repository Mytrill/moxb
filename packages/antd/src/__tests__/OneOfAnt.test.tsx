import { OneOfImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { OneOfAnt, OneOfFormAnt, OneOfSelectAnt, OneOfSelectFormAnt } from '../OneOfAnt';

const allChoices = [{ label: 'Banana', value: 'b' }, { label: 'Apples', value: 'a' }, { label: 'Peaches', value: 'p' }];

describe('OneOfAnt', function() {
    it('should render a radio button group by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallow(<OneOfAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfFormAnt', function() {
    it('should render a radio button group by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallow(<OneOfFormAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfFormAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfSelectAnt', function() {
    it('should render a select control by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallow(<OneOfSelectAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfSelectAnt operation={operation} />).type()).toBeNull();
    });
});

describe('OneOfAnt', function() {
    it('should render a select control by default', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
        });
        expect(shallow(<OneOfSelectFormAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const operation = new OneOfImpl({
            id: 'id.testRadioOfOne',
            label: 'Select one of',
            placeholder: '...',
            choices: () => allChoices,
            invisible: () => true,
        });
        expect(shallow(<OneOfSelectFormAnt operation={operation} />).type()).toBeNull();
    });
});
