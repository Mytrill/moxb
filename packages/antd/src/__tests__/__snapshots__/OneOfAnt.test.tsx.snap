// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneOfAnt should render a radio button group by default 1`] = `
<RadioGroup
  buttonStyle="outline"
  disabled={false}
  id="id-test_radio_of_one"
  label="Select one of"
  onChange={[Function]}
  prefixCls="ant-radio"
  readOnly={false}
>
  <Radio
    checked={false}
    key="b"
    prefixCls="ant-radio"
    type="radio"
    value="b"
  >
    Banana
  </Radio>
  <Radio
    checked={false}
    key="a"
    prefixCls="ant-radio"
    type="radio"
    value="a"
  >
    Apples
  </Radio>
  <Radio
    checked={false}
    key="p"
    prefixCls="ant-radio"
    type="radio"
    value="p"
  >
    Peaches
  </Radio>
</RadioGroup>
`;

exports[`OneOfAnt should render a select control by default 1`] = `
<FormItemAnt
  operation={
    bound OneOfImpl {
      "_errors": Array [],
      "_isSaving": false,
      "_value": undefined,
      "constructor": [Function],
      "doClearErrors": [Function],
      "doGetValue": [Function],
      "doOnValidate": [Function],
      "doPlaceholder": [Function],
      "doSetValue": [Function],
      "domId": "id-test_radio_of_one",
      "getDisabled": [Function],
      "getErrors": [Function],
      "getHelp": [Function],
      "getInitialValue": [Function],
      "getInvisible": [Function],
      "getLabel": [Function],
      "getReadonly": [Function],
      "getReason": [Function],
      "id": "id.testRadioOfOne",
      "impl": Object {
        "choices": [Function],
        "id": "id.testRadioOfOne",
        "label": "Select one of",
        "placeholder": "...",
      },
      "setError": [Function],
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "OneOfImpl@109.keys",
          "observers": Set {},
        },
        "name": "OneOfImpl@109",
        "target": [Circular],
        "values": Map {
          "_errors" => Array [],
          "label" => "Select one of",
          "disabled" => false,
          "enabled" => true,
          "reason" => undefined,
          "invisible" => false,
          "readOnly" => false,
          "help" => undefined,
          "errors" => Array [],
          "error" => undefined,
          "hasErrors" => false,
          "_value" => undefined,
          "_isSaving" => false,
          "inputType" => undefined,
          "value" => undefined,
          "placeholder" => "...",
          "required" => undefined,
          "isInitialValue" => undefined,
          "choices" => Array [
            Object {
              "label": "Banana",
              "value": "b",
            },
            Object {
              "label": "Apples",
              "value": "a",
            },
            Object {
              "label": "Peaches",
              "value": "p",
            },
          ],
        },
      },
    }
  }
>
  <OneOfSelectAnt
    disabled={false}
    id="id-test_radio_of_one"
    operation={
      bound OneOfImpl {
        "_errors": Array [],
        "_isSaving": false,
        "_value": undefined,
        "constructor": [Function],
        "doClearErrors": [Function],
        "doGetValue": [Function],
        "doOnValidate": [Function],
        "doPlaceholder": [Function],
        "doSetValue": [Function],
        "domId": "id-test_radio_of_one",
        "getDisabled": [Function],
        "getErrors": [Function],
        "getHelp": [Function],
        "getInitialValue": [Function],
        "getInvisible": [Function],
        "getLabel": [Function],
        "getReadonly": [Function],
        "getReason": [Function],
        "id": "id.testRadioOfOne",
        "impl": Object {
          "choices": [Function],
          "id": "id.testRadioOfOne",
          "label": "Select one of",
          "placeholder": "...",
        },
        "setError": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "OneOfImpl@109.keys",
            "observers": Set {},
          },
          "name": "OneOfImpl@109",
          "target": [Circular],
          "values": Map {
            "_errors" => Array [],
            "label" => "Select one of",
            "disabled" => false,
            "enabled" => true,
            "reason" => undefined,
            "invisible" => false,
            "readOnly" => false,
            "help" => undefined,
            "errors" => Array [],
            "error" => undefined,
            "hasErrors" => false,
            "_value" => undefined,
            "_isSaving" => false,
            "inputType" => undefined,
            "value" => undefined,
            "placeholder" => "...",
            "required" => undefined,
            "isInitialValue" => undefined,
            "choices" => Array [
              Object {
                "label": "Banana",
                "value": "b",
              },
              Object {
                "label": "Apples",
                "value": "a",
              },
              Object {
                "label": "Peaches",
                "value": "p",
              },
            ],
          },
        },
      }
    }
    readOnly={false}
  />
</FormItemAnt>
`;

exports[`OneOfFormAnt should render a radio button group by default 1`] = `
<FormItemAnt
  operation={
    bound OneOfImpl {
      "_errors": Array [],
      "_isSaving": false,
      "_value": undefined,
      "constructor": [Function],
      "doClearErrors": [Function],
      "doGetValue": [Function],
      "doOnValidate": [Function],
      "doPlaceholder": [Function],
      "doSetValue": [Function],
      "domId": "id-test_radio_of_one",
      "getDisabled": [Function],
      "getErrors": [Function],
      "getHelp": [Function],
      "getInitialValue": [Function],
      "getInvisible": [Function],
      "getLabel": [Function],
      "getReadonly": [Function],
      "getReason": [Function],
      "id": "id.testRadioOfOne",
      "impl": Object {
        "choices": [Function],
        "id": "id.testRadioOfOne",
        "label": "Select one of",
        "placeholder": "...",
      },
      "setError": [Function],
      Symbol(mobx did run lazy initializers): true,
      Symbol(mobx administration): ObservableObjectAdministration$$1 {
        "defaultEnhancer": [Function],
        "keysAtom": Atom$$1 {
          "diffValue": 0,
          "isBeingObserved": false,
          "isPendingUnobservation": false,
          "lastAccessedBy": 0,
          "lowestObserverState": 2,
          "name": "OneOfImpl@37.keys",
          "observers": Set {},
        },
        "name": "OneOfImpl@37",
        "target": [Circular],
        "values": Map {
          "_errors" => Array [],
          "label" => "Select one of",
          "disabled" => false,
          "enabled" => true,
          "reason" => undefined,
          "invisible" => false,
          "readOnly" => false,
          "help" => undefined,
          "errors" => Array [],
          "error" => undefined,
          "hasErrors" => false,
          "_value" => undefined,
          "_isSaving" => false,
          "inputType" => undefined,
          "value" => undefined,
          "placeholder" => "...",
          "required" => undefined,
          "isInitialValue" => undefined,
          "choices" => Array [
            Object {
              "label": "Banana",
              "value": "b",
            },
            Object {
              "label": "Apples",
              "value": "a",
            },
            Object {
              "label": "Peaches",
              "value": "p",
            },
          ],
        },
      },
    }
  }
>
  <OneOfAnt
    disabled={false}
    id="id-test_radio_of_one"
    operation={
      bound OneOfImpl {
        "_errors": Array [],
        "_isSaving": false,
        "_value": undefined,
        "constructor": [Function],
        "doClearErrors": [Function],
        "doGetValue": [Function],
        "doOnValidate": [Function],
        "doPlaceholder": [Function],
        "doSetValue": [Function],
        "domId": "id-test_radio_of_one",
        "getDisabled": [Function],
        "getErrors": [Function],
        "getHelp": [Function],
        "getInitialValue": [Function],
        "getInvisible": [Function],
        "getLabel": [Function],
        "getReadonly": [Function],
        "getReason": [Function],
        "id": "id.testRadioOfOne",
        "impl": Object {
          "choices": [Function],
          "id": "id.testRadioOfOne",
          "label": "Select one of",
          "placeholder": "...",
        },
        "setError": [Function],
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "OneOfImpl@37.keys",
            "observers": Set {},
          },
          "name": "OneOfImpl@37",
          "target": [Circular],
          "values": Map {
            "_errors" => Array [],
            "label" => "Select one of",
            "disabled" => false,
            "enabled" => true,
            "reason" => undefined,
            "invisible" => false,
            "readOnly" => false,
            "help" => undefined,
            "errors" => Array [],
            "error" => undefined,
            "hasErrors" => false,
            "_value" => undefined,
            "_isSaving" => false,
            "inputType" => undefined,
            "value" => undefined,
            "placeholder" => "...",
            "required" => undefined,
            "isInitialValue" => undefined,
            "choices" => Array [
              Object {
                "label": "Banana",
                "value": "b",
              },
              Object {
                "label": "Apples",
                "value": "a",
              },
              Object {
                "label": "Peaches",
                "value": "p",
              },
            ],
          },
        },
      }
    }
    readOnly={false}
  />
</FormItemAnt>
`;

exports[`OneOfSelectAnt should render a select control by default 1`] = `
<Select
  choiceTransitionName="zoom"
  disabled={false}
  id="id-test_radio_of_one"
  label="Select one of"
  mode="default"
  onChange={[Function]}
  prefixCls="ant-select"
  readOnly={false}
  showSearch={false}
  transitionName="slide-up"
  value=""
>
  <Option
    key="b"
    value="b"
  >
    Banana
  </Option>
  <Option
    key="a"
    value="a"
  >
    Apples
  </Option>
  <Option
    key="p"
    value="p"
  >
    Peaches
  </Option>
</Select>
`;
